src/box/lua/net_box.c

```diff
@@ -1735,6 +1735,13 @@ luaT_netbox_request_gc(struct lua_State *L)
 	return 0;
 }
 
+static int
+luaT_netbox_request_autocomplete(struct lua_State *L)
+{
+	lua_getmetatable(L, -1);
+	return 1;
```

Comment:
This isn't all we need to do AFAIU.

It's possible to add userdata to a future object:

```lua
local f = f = c:call('foo', {}, {is_async = true})
f.bar = 123
f.bar -- 123
```

We want user keys to be auto-completed as well.

Reply:
Ok, I made it through the adding an `__autocomplete` metamethod at the moment of new `netbox_request` got it new index table (in fact, only after user value is added). So that the `netbox_request` type will perform the serialization of the index itself, and the metatable of the latter will have chained `__autocomplete` that will deliver the type's completion.